backend/inspirehep/records/indexer/tasks.py

```diff
@@ -81,6 +89,23 @@ def index_record(self, uuid, record_version=None, force_delete=None):
                 attempts=self.max_retries,
             )
         raise self.retry(countdown=backoff, exc=e)
+    except (
+        DisconnectionError,
+        TimeoutError,
+        InvalidatePoolError,
+        OperationalError,
+        UnboundExecutionError,
+        ResourceClosedError,
+    ) as e:
```

Comment:
isn't there a root exception the other inherit from and that you could catch?

Reply:
Only one which is common for all of them is `SQLAlchemyError`
For `DisconnectionError` and `InvalidatePoolError` I could use only `DisconnectionError` as `InvalidatePoolError` is a subclass `DisconnectionError`. But for other exceptions it's not possible :/